{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "u'/Users/gunnarkleemann/Dropbox/coursework/BerkeleyDataSciences_8_11/BerkeleyCODE/W209code/GenomeWeb/PrePrc'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "/pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "if __name__ == '__main__':\n",
    "\n",
    "## dictionaries for recoding \n",
    "# for clinical significance\n",
    "    ClinSigDict={0:99,1:8,2:1,3:2,4:-1,5:-2,6:6,7:7,255:0}\n",
    "    \n",
    "    #DisNmDict={}\n",
    "\n",
    "\n",
    "##FUNCTIONS\n",
    "\n",
    "\n",
    "########################################\n",
    "\n",
    "# Melt data on column names list to serve as Id and value (melted) vars\n",
    "# remove cols with zero values \n",
    "def Melton(splitCol, data):\n",
    "    id_vars=data.columns.values.tolist()[0:splitCol]\n",
    "    value_vars=data.columns.values.tolist()[splitCol:data.shape[1]]\n",
    "    Melted=pd.melt(data, id_vars, value_vars)\n",
    "    print 'lines before empty cell strip:' + str(len(Melted))\n",
    "    Meltedfilt=Melted[Melted['value']==1]\n",
    "    print 'lines After empty cell strip:' + str(len(Meltedfilt))\n",
    "    Meltedfilt.to_csv('genomicDt.csv', sep='\\t', encoding='utf-8', index=False)\n",
    "    return Meltedfilt\n",
    "\n",
    "########################################\n",
    "\n",
    "#MAKE COLUMN NAMES UNIFORM # fix column names,\n",
    "def fixDATA(df, useAlt):\n",
    "    if '#CHROM' in df.columns:\n",
    "        df.rename(columns={'#CHROM': 'CHROM'}, inplace=True)\n",
    "\n",
    "    # remove the mutation address column if it is already there\n",
    "    if 'Mut' in df.columns:\n",
    "        df.drop([\"Mut\"], axis=1, inplace=True)\n",
    "\n",
    "    #make all chrom titles uniform\n",
    "    try:\n",
    "        if len(df[\"CHROM\"][1]) > 4:\n",
    "            df[\"CHROM\"] = df[\"CHROM\"].apply(lambda x: x[3:len(x)])\n",
    "    except:\n",
    "        print \"chr no len\"\n",
    "\n",
    "    # make mutation address column\n",
    "    chr = df[\"CHROM\"].map(str)\n",
    "    #print df[\"POS\"]\n",
    "    pos = df[\"POS\"].map(str)\n",
    "\n",
    "    print useAlt\n",
    "\n",
    "    if useAlt == 'yes':\n",
    "        alt = df[\"ALT\"].map(str)\n",
    "        df['Mut'] = 'chr' + chr + '-' + pos + '-' + alt\n",
    "        print 'added alt'\n",
    "    else:\n",
    "        df['Mut'] = 'chr' + chr + '-' + pos\n",
    "        print 'added NO alt'\n",
    "    return df\n",
    "\n",
    "########################################\n",
    "\n",
    "def FirstItem(str, sep):\n",
    "    Item=str.split(sep)[0]\n",
    "    return Item\n",
    "\n",
    "#############GetInfo, Get 1 datum from info field#######################\n",
    "\n",
    "### df=FiltList # this is the dataframe\n",
    "### requires that the queried column is called'INFO' \n",
    "### fieldNm ='GENEINFO' #This is the field in the that we want to capture#\n",
    "### ColNm = 'gene' #This is the name of column to put data into\n",
    "\n",
    "#THE INFO FIELDS ARE NOT IN CONSTANT PLACES\n",
    "#to find the right field, cycle through the elements of each list - the field changes position!\n",
    "#NOTE: the info fields can contain repeat fields\n",
    "#For the sake of this build I am taking the first\n",
    "#Info = FiltList.INFO_y[n].split(';') # Parse Info data\n",
    "#InfoNm needs to be dynamically assigned\n",
    "\n",
    "def GetInfo(df, fieldNm, ColNm, InfoNm):    \n",
    "    for n in range(len(df) - 1):   # for each line get the info\n",
    "        Info = df.INFO_y[n].split(';') \n",
    "        Field = 'NotListed' # assume it is not there until its found\n",
    "        for p in range(len(Info)): # cycle through info field for correct tag\n",
    "            if Info[p][0:len(fieldNm)] == fieldNm: # parse data and put in the array\n",
    "                str1=Info[p][len(fieldNm)+1:len(Info[p])]\n",
    "                str2=FirstItem(str1, ':') #Remove number and take first item\n",
    "                try:\n",
    "                    Field=FirstItem(str2, '|')\n",
    "                except:\n",
    "                    pass\n",
    "                try:\n",
    "                    Field=FirstItem(str2, ',')\n",
    "                except:\n",
    "                    pass\n",
    "            else:\n",
    "                next\n",
    "        df.loc[n, ColNm] = Field   \n",
    "    return df\n",
    "\n",
    "########################################\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#Extract and clean the Clincal severity (CLNSIG), \n",
    "#and gene info (GENINFO) from the INFO columns\n",
    "#InfoNm needs to be dynamically assigned\n",
    "def GetClinDt(FiltList, InfoNm):\n",
    "    for n in range(len(FiltList) - 1):\n",
    "            Info = FiltList.INFO_y[n].split(';') # Parse Info data\n",
    "            FiltList.loc[n, ('SnpDB')] = 'http://www.ncbi.nlm.nih.gov/snp/' + Info[0][3:len(Info[0])]  #gen the dbSNP link\n",
    "            FiltList.loc[n, ('RSnum')] = Info[0][3:len(Info[0])]\n",
    "\n",
    "    FiltList=GetInfo(FiltList, 'CLNSIG', 'ClinSig',InfoNm)  \n",
    "    FiltList=GetInfo(FiltList, 'GENEINFO', 'Gene',InfoNm)    \n",
    "    FiltList=GetInfo(FiltList, 'CLNDBN', 'Disease', InfoNm)                  \n",
    "    return FiltList\n",
    "        \n",
    "# Transform the signficance values\n",
    "def recode(SignifVal, RecodeDict):\n",
    "    # for a tricolor (and black) scale...\n",
    "    #0 - Uncertain significance (keep 0 middle value)\n",
    "    #1 - not provided, (recode 255 black)\n",
    "    #2 - Benign, (recode 1)\n",
    "    #3 - Likely benign, (recode 2)\n",
    "    #4 - Likely pathogenic,  (recode -1)\n",
    "    #5 - Pathogenic, (recode -2)\n",
    "    #6 - drug response(?)\n",
    "    #7 - histocompatibility(?)\n",
    "    #255 - \"other\" recode (0) unknown\n",
    "    #RecodeDict={0:0,1:255,2:1,3:2,4:-1,5:-2,6:6,7:7} # pass this into function\n",
    "    recodeVal=RecodeDict[SignifVal]\n",
    "    return (recodeVal)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#IMPORT THE DATA - this method pulls lines from the s3 files. let try something \n",
    "#easier for now\n",
    "#import numpy as np\n",
    "#import pandas as pd\n",
    "#import urllib\n",
    "\n",
    "\n",
    "## these files are also on S3\n",
    "## https://s3-us-west-2.amazonaws.com/209genomics/20130606_sample_info2.csv\n",
    "## https://s3-us-west-2.amazonaws.com/209genomics/ALL.chrMT.phase3_callmom.20130502.genotypes.vcf.txt\n",
    "## https://s3-us-west-2.amazonaws.com/209genomics/clinvar_20150804.vcf.txt\n",
    "\n",
    "\n",
    "#opener = urllib.URLopener()\n",
    "#myurl = \"https://s3-us-west-2.amazonaws.com/209genomics/20130606_sample_info2.csv\"\n",
    "#myfile = opener.open(myurl)\n",
    "#SampleInfo = myfile.read()\n",
    "#SampleInfo=pd.read_table(SampleInfo, sep=',')\n",
    "#print SampleInfo\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import urllib\n",
    "\n",
    "Genomes='ALL.chrMT.phase3_callmom.20130502.genotypes.vcf.txt'\n",
    "#MutRef='common_and_clinical.vcf.txt' # this one has no Mitochondrial X and Y data\n",
    "SampleInfo='20130606_sample_info2.csv'\n",
    "MutRef2='clinvar_20150804.vcf.txt' # this one HAS Mitochondrial X and Y data\n",
    "\n",
    "SkipCt1=9\n",
    "hdCt2=55\n",
    "hdCt3=66\n",
    "\n",
    "SampleInfo=pd.read_table(SampleInfo, sep=',')\n",
    "Genomes=pd.read_table(Genomes, skiprows=SkipCt1)\n",
    "#MutRef=pd.read_table(MutRef, skiprows=hdCt2, low_memory=False)\n",
    "MutRef2=pd.read_table(MutRef2, skiprows=hdCt3, low_memory=False)\n",
    "# the data from the genome files is wide format they have to be melted to make this work"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "lines before empty cell strip:9816716\n",
      "lines After empty cell strip:89695\n"
     ]
    }
   ],
   "source": [
    "# melt the data and strip empty cells - \n",
    "GenomesMelt=Melton(9, Genomes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yes\n",
      "added alt\n",
      "yes\n",
      "added alt\n",
      "chr no len\n",
      "yes\n",
      "added alt\n"
     ]
    }
   ],
   "source": [
    "#fix the names and add the MUT colum to use for joining\n",
    "useAlt='yes'\n",
    "\n",
    "MutFile_fx = fixDATA(MutRef2, useAlt) # use this file for MT data\n",
    "Genomes_fx = fixDATA(Genomes, useAlt)\n",
    "GenomesMelt_fx = fixDATA(GenomesMelt, useAlt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([u'POS', u'HG00096', u'HG00097', u'HG00099', u'HG00100', u'HG00101',\n",
       "       u'HG00102', u'HG00103', u'HG00105', u'HG00106', \n",
       "       ...\n",
       "       u'NA21129', u'NA21130', u'NA21133', u'NA21135', u'NA21137', u'NA21141',\n",
       "       u'NA21142', u'NA21143', u'NA21144', u'Mut'],\n",
       "      dtype='object', length=2536)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#We need to put Genomes_fx into a file suitable for clustering \n",
    "# row ID is variant, column ID is person \n",
    "\n",
    "# one format is .PCL file (preclustering) suitable for Java treeview to cluster. \n",
    "# a simpler format is for clustering in R or using\n",
    "Genomes_fx.drop(['CHROM','QUAL','ALT','REF','FILTER','INFO','FORMAT','ID'], axis=1, inplace=True)\n",
    "Genomes_fx.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/IPython/kernel/__main__.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mut</th>\n",
       "      <th>HG00096</th>\n",
       "      <th>HG00097</th>\n",
       "      <th>HG00099</th>\n",
       "      <th>HG00100</th>\n",
       "      <th>HG00101</th>\n",
       "      <th>HG00102</th>\n",
       "      <th>HG00103</th>\n",
       "      <th>HG00105</th>\n",
       "      <th>HG00106</th>\n",
       "      <th>...</th>\n",
       "      <th>NA21128</th>\n",
       "      <th>NA21129</th>\n",
       "      <th>NA21130</th>\n",
       "      <th>NA21133</th>\n",
       "      <th>NA21135</th>\n",
       "      <th>NA21137</th>\n",
       "      <th>NA21141</th>\n",
       "      <th>NA21142</th>\n",
       "      <th>NA21143</th>\n",
       "      <th>NA21144</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrMT-10-C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrMT-16-T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 2535 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Mut  HG00096  HG00097  HG00099  HG00100  HG00101  HG00102  HG00103  \\\n",
       "0  chrMT-10-C        0        0        0        0        0        0        0   \n",
       "1  chrMT-16-T        0        0        0        0        0        0        0   \n",
       "\n",
       "   HG00105  HG00106   ...     NA21128  NA21129  NA21130  NA21133  NA21135  \\\n",
       "0        0        0   ...           0        0        0        0        0   \n",
       "1        0        0   ...           0        0        0        0        0   \n",
       "\n",
       "   NA21137  NA21141  NA21142  NA21143  NA21144  \n",
       "0        0        0        0        0        0  \n",
       "1        0        0        0        0        0  \n",
       "\n",
       "[2 rows x 2535 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#clean up the HEATMAP MATRIX file\n",
    "\n",
    "cols = Genomes_fx.columns.tolist()\n",
    "\n",
    "Pos=Genomes_fx.iloc[:,0:1]\n",
    "\n",
    "cols = cols[-1:] + cols[:-1]# move the mutation column into the first position\n",
    "\n",
    "Genomes_fx=Genomes_fx[cols]\n",
    "\n",
    "Genomes_fx.drop(['POS'], axis=1, inplace=True)\n",
    "\n",
    "#Genomes_fx=Pos + Genomes_fx \n",
    "\n",
    "Genomes_fx.head(2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>Mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>rs786201005</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=786201005;RSPOS=1014143;dbSNPBuildID=144;SS...</td>\n",
       "      <td>chr1-1014143-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014316</td>\n",
       "      <td>rs672601345</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>RS=672601345;RSPOS=1014319;dbSNPBuildID=142;SS...</td>\n",
       "      <td>chr1-1014316-CG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS           ID REF ALT QUAL FILTER  \\\n",
       "0     1  1014143  rs786201005   C   T    .      .   \n",
       "1     1  1014316  rs672601345   C  CG    .      .   \n",
       "\n",
       "                                                INFO              Mut  \n",
       "0  RS=786201005;RSPOS=1014143;dbSNPBuildID=144;SS...   chr1-1014143-T  \n",
       "1  RS=672601345;RSPOS=1014319;dbSNPBuildID=142;SS...  chr1-1014316-CG  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Use the fixed clinvar file (MutFile) to get to the disease, clinical significance and gene information\n",
    "#MutFile_fx.to_csv('ClinVar_fx.txt')\n",
    "MutFile_fx.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Family ID</th>\n",
       "      <th>Population</th>\n",
       "      <th>Population Description</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Relationship</th>\n",
       "      <th>Unexpected Parent/Child</th>\n",
       "      <th>Non Paternity</th>\n",
       "      <th>Siblings</th>\n",
       "      <th>Grandparents</th>\n",
       "      <th>Avuncular</th>\n",
       "      <th>Half Siblings</th>\n",
       "      <th>Unknown Second Order</th>\n",
       "      <th>Third Order</th>\n",
       "      <th>Other Comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00096</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Sample Family ID Population           Population Description Gender  \\\n",
       "0  HG00096   HG00096        GBR  British in England and Scotland   male   \n",
       "\n",
       "  Relationship Unexpected Parent/Child  Non Paternity Siblings Grandparents  \\\n",
       "0          NaN                      NaN           NaN      NaN          NaN   \n",
       "\n",
       "  Avuncular Half Siblings Unknown Second Order Third Order  Other Comments  \n",
       "0       NaN           NaN                  NaN         NaN             NaN  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SampleInfo.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "lines before Clinvar filter:3874\n",
      "lines After ClinVar filter:111\n",
      "111 Unique Mutations\n"
     ]
    }
   ],
   "source": [
    "##FILTER FILES FOR KNOWN MUTATIONS\n",
    "## Merge both INDIVIDUAL AND MATRIX data on the list of known mutations (ClivarList = MutFile_fx )\n",
    "## This filter removes mutations that we have no information about and \n",
    "## puts the gene, clinical significance and gene data in register with the mutation (MUT) Column.\n",
    "\n",
    "print 'lines before Clinvar filter:' + str(len(Genomes_fx))\n",
    "#The heatmap data\n",
    "FiltList = pd.merge(Genomes_fx, MutFile_fx, on=['Mut'], how='inner')\n",
    "#The individual genome summary\n",
    "FiltListMelt = pd.merge(GenomesMelt_fx, MutFile_fx, on=['Mut'], how='inner')\n",
    "print 'lines After ClinVar filter:' + str(len(FiltList))\n",
    "print str(len(np.unique(FiltList['Mut']))) + ' Unique Mutations'\n",
    "#print ('\\n')\n",
    "#print 'heatmap (FiltList)'\n",
    "#print ('\\n')\n",
    "#print FiltList.head(3)\n",
    "#print ('\\n')\n",
    "#print 'Individual summary (FiltListMelt)'\n",
    "#print ('\\n')\n",
    "#print FiltListMelt.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(111, 2543)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## WE MAY OPT TO REMOVE COLUMNS/ ROWS where the \"mutation\" \n",
    "## is found in most patients and therefore not a mutation.\n",
    "FiltList.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index([u'CHROM_x', u'POS_x', u'REF_x', u'ALT_x', u'INFO_x', u'Sample',\n",
      "       u'value', u'Mut', u'CHROM_y', u'POS_y', u'REF_y', u'ALT_y', u'INFO_y',\n",
      "       u'Family ID', u'Population', u'Population Description', u'Gender'],\n",
      "      dtype='object')\n",
      "(10424, 17)\n"
     ]
    }
   ],
   "source": [
    "#Merge the sample info with INDIVUDUAL GENOME SUMMARY\n",
    "#list on the genome(variable/Sample) feature and clean up\n",
    "\n",
    "FiltListMelt.rename(columns={'variable':'Sample'}, inplace=True)\n",
    "FiltListMelt = pd.merge(FiltListMelt, SampleInfo, on=['Sample'], how='inner')\n",
    "FiltListMelt.drop(FiltListMelt.columns[24:len(FiltListMelt.columns)], axis=1, inplace= True)\n",
    "\n",
    "#FiltList.drop(FiltList.columns[[2,17,18]], axis=1, inplace= True)\n",
    "#FiltList.drop(FiltList.columns[[5,7,9]], axis=1, inplace= True)\n",
    "FiltListMelt.drop(['ID_x','QUAL_x', 'FILTER_x', 'FORMAT','ID_y','QUAL_y', 'FILTER_y'], axis=1, inplace= True)\n",
    "\n",
    "print FiltListMelt.columns\n",
    "print FiltListMelt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index([u'CHROM_x', u'POS_x', u'REF_x', u'ALT_x', u'INFO_x', u'Sample',\n",
      "       u'value', u'Mut', u'CHROM_y', u'POS_y', u'REF_y', u'ALT_y', u'INFO_y',\n",
      "       u'Family ID', u'Population', u'Population Description', u'Gender',\n",
      "       u'SnpDB', u'RSnum', u'ClinSig', u'Gene', u'Disease'],\n",
      "      dtype='object')\n",
      "(10424, 22)\n",
      "             Mut  HG00096  HG00097  HG00099  HG00100  HG00101  HG00102  \\\n",
      "0    chrMT-827-G        0        0        0        0        0        0   \n",
      "1    chrMT-951-A        0        0        0        0        0        0   \n",
      "2  chrMT-961-C,G        0        0        0        0        0        0   \n",
      "3    chrMT-980-C        0        0        0        0        0        0   \n",
      "4    chrMT-990-C        0        0        0        0        0        0   \n",
      "\n",
      "   HG00103  HG00105  HG00106  \\\n",
      "0        0        0        0   \n",
      "1        0        0        0   \n",
      "2        0        0        0   \n",
      "3        0        0        0   \n",
      "4        0        0        0   \n",
      "\n",
      "                         ...                          NA21143  NA21144  CHROM  \\\n",
      "0                        ...                                0        0     MT   \n",
      "1                        ...                                0        0     MT   \n",
      "2                        ...                                0        0     MT   \n",
      "3                        ...                                0        0     MT   \n",
      "4                        ...                                0        0     MT   \n",
      "\n",
      "   POS           ID  REF  ALT  QUAL  FILTER  \\\n",
      "0  827   rs28358569    A    G     .       .   \n",
      "1  951  rs200887992    G    A     .       .   \n",
      "2  961    rs3888511    T  C,G     .       .   \n",
      "3  980  rs397515731    T    C     .       .   \n",
      "4  990  rs727503165    T    C     .       .   \n",
      "\n",
      "                                                INFO  \n",
      "0  RS=28358569;RSPOS=827;dbSNPBuildID=125;SSR=0;S...  \n",
      "1  RS=200887992;RSPOS=951;dbSNPBuildID=136;SSR=0;...  \n",
      "2  RS=3888511;RSPOS=961;dbSNPBuildID=108;SSR=0;SA...  \n",
      "3  RS=397515731;RSPOS=980;dbSNPBuildID=136;SSR=0;...  \n",
      "4  RS=727503165;RSPOS=990;dbSNPBuildID=142;SSR=0;...  \n",
      "\n",
      "[5 rows x 2543 columns]\n"
     ]
    }
   ],
   "source": [
    "##FOR THE INDIVIDUALS FILE - ADD the DISEASE DATA AND SAVE\n",
    "#FiltListMelt.to_csv('1000genomesMtData')\n",
    "FiltList2=GetClinDt(FiltListMelt,'INFO_y')\n",
    "\n",
    "#OUTPUT THE INDIVIDUAL DATA\n",
    "out='INDIVgenomicDt'\n",
    "datasrc='Mt'\n",
    "Align='ALL.chrMT.phase3_callmom.20130502.genotypes.vcf.txt'\n",
    "ref='common_and_clinical.vcf.txt'\n",
    "fileNm = out + '_' + datasrc + '.csv'\n",
    "FiltList2.to_csv(fileNm)\n",
    "\n",
    "print FiltList2.columns\n",
    "print FiltList2.shape\n",
    "print FiltList.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found a NaN on line10423\n"
     ]
    }
   ],
   "source": [
    "##Add new column to Individual file to recode DISEASE NAMES to simple IDS.\n",
    "\n",
    "FiltList2['DiseaseSumm']=FiltList2['Disease']\n",
    "FiltList2.head()\n",
    "\n",
    "\n",
    "dlst=FiltList2.groupby('Disease')\n",
    "Dlist=dlst.Disease.unique()\n",
    "Dlist.to_csv(\"diseaseList.csv\")\n",
    "Dlist\n",
    "\n",
    "## lets make a dictionary to be used to simpilfy terms##\n",
    "diseaseDict ={'Aminoglycoside-induced_deafness|Deafness\\\\x2c_nonsyndromic_sensorineural\\\\x2c_mitochondrial|Auditory_neuropathy|not_specified': 'Deafness',  \n",
    "'Brain_pseudoatrophy\\\\x2c_reversible\\\\x2c_valproate-induced\\\\x2c_susceptibility_to': 'Brain_pseudoatrophy',\n",
    "'Cardiomyopathy\\\\x2c_idiopathic_dilated\\\\x2c_mitochondrial|Cardiomyopathy\\\\x2c_hypertrophic\\\\x2c_mitochondrial': 'Cardiomyopathy',\n",
    "'Cardiomyopathy\\\\x2c_mitochondrial': 'Cardiomyopathy',\n",
    "'Cytochrome-c_oxidase_deficiency': 'Cytochrome-c deficiency',\n",
    "'Deafness\\\\x2c_nonsyndromic_sensorineural\\\\x2c_mitochondrial': 'Deafness',\n",
    "'Deafness\\\\x2c_nonsyndromic_sensorineural\\\\x2c_mitochondrial|Aminoglycoside-induced_deafness': 'Deafness',\n",
    "'Familial_cancer_of_breast': 'Breast cancer',\n",
    "'Familial_colorectal_cancer': 'colorectal cancer',\n",
    "'Focal_segmental_glomerulosclerosis_and_dilated_cardiomyopathy': 'Glomerulosclerosis and Cardiomyopathy',\n",
    "'Juvenile_myopathy\\\\x2c_encephalopathy\\\\x2c_lactic_acidosis_AND_stroke': 'Juvenile_myopathy...',\n",
    "\"Leber's_optic_atrophy\": 'Lebers_optic_atrophy',\n",
    "\"Leber's_optic_atrophy|Aminoglycoside-induced_deafness|Deafness\\\\x2c_nonsyndromic_sensorineural\\\\x2c_mitochondrial\": 'Lebers_optic_atrophy',\n",
    "\"Leber_hereditary_optic_neuropathy_with_dystonia|Leber's_optic_atrophy\": 'Lebers_optic_atrophy',\n",
    "\"Leigh's_disease\": 'Leighs_disease',\n",
    "'Neoplasm_of_ovary': 'Neoplasm_of_ovary',\n",
    "'Obesity': 'Obesity',\n",
    "'Parkinson_disease\\\\x2c_resistance_to': 'Parkinson_disease',\n",
    "'Parkinson_disease_6\\\\x2c_autosomal_recessive_early-onset': 'Parkinson_disease',\n",
    "'Primary_familial_hypertrophic_cardiomyopathy|Deafness\\\\x2c_nonsyndromic_sensorineural\\\\x2c_mitochondrial': 'cardiomyopathy',\n",
    "'Sensorineural_deafness_and_migraine': 'Deafness',\n",
    "'Sudden_death': 'Sudden death',\n",
    "'Sudden_infant_death_syndrome': 'SIDS',\n",
    "'not_specified': 'not specified'}\n",
    "\n",
    "## now replace the disease names in the new columns with the summary names\n",
    "\n",
    "for n in range(len(FiltList2)):\n",
    "    try :\n",
    "        FiltList2.iloc[n, 22]=diseaseDict[FiltList2.iloc[n, 21]] \n",
    "    except:\n",
    "        print 'Found a NaN on line' + str(n)\n",
    "        FiltList2.iloc[n, 22]='NaN'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "##NOW FOCUS ON THE HEATMAP MATRIX## \n",
    "# Recode called mutations with thier clinical significance and \n",
    "# EITHER APPEND OR generate a second dataframe with disease and gene names. \n",
    "\n",
    "# The internal values are clincal significance and should be coded more intuitively\n",
    "# ##INFO=<ID=CLNSIG,Number=.,Type=String,Description=\"Variant Clinical Significance, \n",
    "\n",
    "# for a tricolor scale...\n",
    "#0 - Uncertain significance (keep 0 middle value)\n",
    "#1 - not provided, (recode 255 black)\n",
    "#2 - Benign, (recode 1)\n",
    "#3 - Likely benign, (recode 2)\n",
    "#4 - Likely pathogenic,  (recode -1)\n",
    "#5 - Pathogenic, (recode -2)\n",
    "#6 - drug response(?)\n",
    "#7 - histocompatibility(?) \n",
    "# need to recode the non-muation sites 255 - Just Drop these \n",
    "## have 6, 7 and 255 been stripped by the fix data function?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Family ID</th>\n",
       "      <th>Population</th>\n",
       "      <th>Population Description</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Relationship</th>\n",
       "      <th>Unexpected Parent/Child</th>\n",
       "      <th>Non Paternity</th>\n",
       "      <th>Siblings</th>\n",
       "      <th>Grandparents</th>\n",
       "      <th>Avuncular</th>\n",
       "      <th>Half Siblings</th>\n",
       "      <th>Unknown Second Order</th>\n",
       "      <th>Third Order</th>\n",
       "      <th>Other Comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3495</th>\n",
       "      <td>NA21137</td>\n",
       "      <td>NA21137</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3496</th>\n",
       "      <td>NA21141</td>\n",
       "      <td>NA21141</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3497</th>\n",
       "      <td>NA21142</td>\n",
       "      <td>NA21142</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3498</th>\n",
       "      <td>NA21143</td>\n",
       "      <td>NA21143</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3499</th>\n",
       "      <td>NA21144</td>\n",
       "      <td>NA21144</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sample Family ID Population         Population Description  Gender  \\\n",
       "3495  NA21137   NA21137        GIH  Gujarati Indian in Houston,TX  female   \n",
       "3496  NA21141   NA21141        GIH  Gujarati Indian in Houston,TX  female   \n",
       "3497  NA21142   NA21142        GIH  Gujarati Indian in Houston,TX  female   \n",
       "3498  NA21143   NA21143        GIH  Gujarati Indian in Houston,TX  female   \n",
       "3499  NA21144   NA21144        GIH  Gujarati Indian in Houston,TX  female   \n",
       "\n",
       "     Relationship Unexpected Parent/Child  Non Paternity Siblings  \\\n",
       "3495          NaN                      NaN           NaN      NaN   \n",
       "3496          NaN                      NaN           NaN      NaN   \n",
       "3497          NaN                      NaN           NaN      NaN   \n",
       "3498          NaN                      NaN           NaN      NaN   \n",
       "3499          NaN                      NaN           NaN      NaN   \n",
       "\n",
       "     Grandparents Avuncular Half Siblings Unknown Second Order Third Order  \\\n",
       "3495          NaN       NaN           NaN                  NaN         NaN   \n",
       "3496          NaN       NaN           NaN                  NaN         NaN   \n",
       "3497          NaN       NaN           NaN                  NaN         NaN   \n",
       "3498          NaN       NaN           NaN                  NaN         NaN   \n",
       "3499          NaN       NaN           NaN                  NaN         NaN   \n",
       "\n",
       "      Other Comments  \n",
       "3495             NaN  \n",
       "3496             NaN  \n",
       "3497             NaN  \n",
       "3498             NaN  \n",
       "3499             NaN  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SampleInfo.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(111, 2535)\n",
      "[['chrMT-951-A' 0 0 ..., 0 0 0]\n",
      " ['chrMT-961-C,G' 0 0 ..., 0 0 0]\n",
      " ['chrMT-980-C' 0 0 ..., 0 0 0]\n",
      " ['chrMT-990-C' 0 0 ..., 0 0 0]]\n"
     ]
    }
   ],
   "source": [
    "## FIX THE MATRIX \n",
    "\n",
    "\n",
    "##split out the HEATMAP data, retain mutation 'mut' column\n",
    "Mut_Matrix=FiltList.iloc[:,0:2535]\n",
    "\n",
    "    #Mut_Matrix.drop('POS_x', axis=1, inplace=True)    \n",
    "Mut_nparray = Mut_Matrix.as_matrix() # make a numpy array for searching\n",
    "print Mut_nparray.shape\n",
    "print Mut_nparray[1:5,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# FIND THE INDEX OF THE MUTATIONS (non-zeros) and the WT sites (zeros)) \n",
    "# transpose is used to coerce the indexes into a usable shape\n",
    "#NonZeroIdx=np.transpose(np.nonzero(Mut_nparray[:,1:Mut_nparray.shape[1]]))\n",
    "#ZeroIdx=np.transpose(np.where(Mut_nparray == 0))\n",
    "#adjust for the header row.\n",
    "#NonZeroIdx[:,1] = NonZeroIdx[:,1]+1\n",
    "#ZeroIdx[:,1] = ZeroIdx[:,1]+1\n",
    "#print NonZeroIdx[1:5,:]\n",
    "#print ZeroIdx[1:5,:]\n",
    "#print NonZeroIdx[:,1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## RECODE the matrix replacing mutations with thier severity score\n",
    "## keep a copy of the matrix for comparison and to indicate haploid(1) or diploid(2)\n",
    "ClinSigDict={0:99,1:8,2:1,3:2,4:-1,5:-2,6:6,7:7,255:0}\n",
    "Mut_MatrixRECODE=Mut_Matrix.replace(ClinSigDict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "Mut_MatrixRECODE=Mut_MatrixRECODE.transpose()\n",
    "Mut_Matrix=Mut_Matrix.transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## RECODE the matrix replacing mutations with thier severity score\n",
    "## keep a copy of the matrix for comparison and to indicate haploid(1) or diploid(2)\n",
    "\n",
    "\n",
    "# lookup the mutation - Get the severity score and Place it in the matrix \n",
    "#for ii in range(len(NonZeroIdx)):\n",
    "#    MutID=Mut_nparray[NonZeroIdx[ii][0],0] # identify the mutation\n",
    "#    #print MutID + '_' + str(NonZeroIdx[z][0]) + \"_\" + str((NonZeroIdx[z][1])+1)\n",
    "#    row=np.where(FiltListMelt['Mut']==MutID)[0][0]\n",
    "#    ClinSig=int(FirstItem(FiltListMelt.iloc[row,19], '|'))\n",
    "#    #print ClinSig\n",
    "#    ClinSig=recode(ClinSig, ClinSigDict)\n",
    "#    #Mut_Matrix.iloc[NonZeroIdx[ii][0],(NonZeroIdx[ii][1])]=ClinSig\n",
    "#    if (Mut_Matrix.iloc[NonZeroIdx[ii][0],(NonZeroIdx[ii][1])]) == 0: # none of these positions should be 0\n",
    "#        print 'replacing' + str(Mut_Matrix.iloc[NonZeroIdx[ii][0],(NonZeroIdx[ii][1])]) + 'with' + str(ClinSig)\n",
    "#        print \"cell\" + str(NonZeroIdx[ii][1]) + 'and '+  str(NonZeroIdx[ii][0])\n",
    "    \n",
    "#    try:\n",
    "#        Mut_Matrix.iloc[NonZeroIdx[ii][0],(NonZeroIdx[ii][1])]=ClinSig\n",
    "#    except:\n",
    "#        print 'replacing' + str(Mut_Matrix.iloc[NonZeroIdx[ii][0],(NonZeroIdx[ii][1])]) + 'with' + str(ClinSig)\n",
    "#        print \"failed at cell\" + str(NonZeroIdx[ii][1]) + 'x cell '+  str(NonZeroIdx[ii][0])\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Lookup WT alleles get position and code with 255\n",
    "#for j in range(len(ZeroIdx)):\n",
    "    #MutID=Mut_nparray[ZeroIdx[j][0],0]\n",
    "    #print MutID + '_' + str(NonZeroIdx[z][0]) + \"_\" + str((NonZeroIdx[z][1])+1)\n",
    "    #row=np.where(FiltListMelt['Mut']==MutID)[0][0]  \n",
    "    \n",
    "    # none of these positions should be non-0\n",
    "#    if (Mut_Matrix.iloc[ZeroIdx[ii][0],(ZeroIdx[ii][1])]) != 0: \n",
    "#        print 'replacing' + str(Mut_Matrix.iloc[ZeroIdx[ii][0],(ZeroIdx[ii][1])]) + 'with' + str(ClinSig)\n",
    "#        print \"cell\" + str(ZeroIdx[ii][1]) + 'and '+  str(ZeroIdx[ii][0])\n",
    "    \n",
    "#    try:\n",
    "#        Mut_Matrix.iloc[ZeroIdx[j][0],(ZeroIdx[j][1])]=255 #fill all of the WT positions\n",
    "        #print Mut_Matrix.iloc[ZeroIdx[j][0],(ZeroIdx[j][1])]\n",
    "#    except:\n",
    "        # it is missing the last row...\n",
    "#        print ZeroIdx[j][0]\n",
    "#        print ZeroIdx[j][1]\n",
    "        #print j"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'SampleID' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-24-cfeaa70a090a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mSampleID\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'SampleID' is not defined"
     ]
    }
   ],
   "source": [
    "SampleID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "## EXTRACT KEYS AND MATCH WITH PERTINENT DATA\n",
    "\n",
    "\n",
    "### NEED TO ADD DISESES AND GENES HEE\n",
    "#MutNames.columns\n",
    "#MutNames=pd.merge(MutNames, SampleInfo, on=['Sample'], how='inner')\n",
    "#SampleInfo\n",
    "#THIS WILL BE USED TO GET THE GENE/DISEASE INFOR\n",
    "####>MutNames=pd.DataFrame(Mut_Matrix.iloc[0,:], columns=['Mut'])\n",
    "####>MutNames=pd.merge(MutNames, SampleInfo, on=['Sample'], how='inner') \n",
    "####>MutNames = pd.merge(MutNames, MutFile_fx, on=['Mut'], how='inner')\n",
    "####>to_csv()\n",
    "\n",
    "#print MutNames\n",
    "SampleID=pd.DataFrame(Mut_Matrix.index, columns =['Sample'])\n",
    "SampleID=pd.merge(SampleID, SampleInfo, on=['Sample'], how='inner') \n",
    "SampleID=SampleID.iloc[:,1:5]\n",
    "SampleID['PopCode']=SampleID['Population']\n",
    "SampleID['PopCode']=SampleID['PopCode'].replace(['GBR', 'FIN', 'CHS', 'PUR', 'CDX', 'CLM', 'IBS', 'PEL', 'PJL',\n",
    "       'KHV', 'ACB', 'GWD', 'ESN', 'BEB', 'MSL', 'STU', 'ITU', 'CEU',\n",
    "       'YRI', 'CHB', 'JPT', 'LWK', 'ASW', 'MXL', 'TSI', 'GIH'], ['1','2','3','4','5','6','7','8', '9',\n",
    "       '10', '11', '12', '13', '14', '15', '16', '17', '18',\n",
    "       '19', '20', '21', '22', '23', '24', '25', '26'])\n",
    "\n",
    "SampleID.to_csv('SampleID.csv', sep='\\t', encoding='utf-8', index=False)\n",
    "\n",
    "#Mut_MatrixRECODE.head()\n",
    "# attatch the population data to reference array MutMatrix\n",
    "#Mut_Matrix.columns\n",
    "#={'':'Sample'}, inplace=True)\n",
    "#Mut_Matrix.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Family ID</th>\n",
       "      <th>Population</th>\n",
       "      <th>Population Description</th>\n",
       "      <th>Gender</th>\n",
       "      <th>PopCode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00096</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00097</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00100</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00101</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HG00102</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HG00103</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HG00105</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HG00106</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HG00107</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HG00108</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HG00109</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HG00110</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HG00111</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HG00112</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HG00113</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HG00114</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HG00115</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>HG00116</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HG00117</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HG00118</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>HG00119</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HG00120</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>HG00121</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>HG00122</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HG00123</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HG00124</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>HG00125</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>HG00126</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>HG00127</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2504</th>\n",
       "      <td>NA21107</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2505</th>\n",
       "      <td>NA21108</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2506</th>\n",
       "      <td>NA21109</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2507</th>\n",
       "      <td>NA21110</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2508</th>\n",
       "      <td>NA21111</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2509</th>\n",
       "      <td>NA21112</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2510</th>\n",
       "      <td>NA21113</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2511</th>\n",
       "      <td>NA21114</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2512</th>\n",
       "      <td>NA21115</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2513</th>\n",
       "      <td>NA21116</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2514</th>\n",
       "      <td>NA21117</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2515</th>\n",
       "      <td>NA21118</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2516</th>\n",
       "      <td>NA21119</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2517</th>\n",
       "      <td>NA21120</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2518</th>\n",
       "      <td>NA21122</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2519</th>\n",
       "      <td>NA21123</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2520</th>\n",
       "      <td>NA21124</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2521</th>\n",
       "      <td>NA21125</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2522</th>\n",
       "      <td>NA21126</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2523</th>\n",
       "      <td>NA21127</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2524</th>\n",
       "      <td>NA21128</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2525</th>\n",
       "      <td>NA21129</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2526</th>\n",
       "      <td>NA21130</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2527</th>\n",
       "      <td>NA21133</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2528</th>\n",
       "      <td>NA21135</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>male</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2529</th>\n",
       "      <td>NA21137</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2530</th>\n",
       "      <td>NA21141</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2531</th>\n",
       "      <td>NA21142</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2532</th>\n",
       "      <td>NA21143</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2533</th>\n",
       "      <td>NA21144</td>\n",
       "      <td>GIH</td>\n",
       "      <td>Gujarati Indian in Houston,TX</td>\n",
       "      <td>female</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2534 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Family ID Population           Population Description  Gender PopCode\n",
       "0      HG00096        GBR  British in England and Scotland    male       1\n",
       "1      HG00097        GBR  British in England and Scotland  female       1\n",
       "2      HG00099        GBR  British in England and Scotland  female       1\n",
       "3      HG00100        GBR  British in England and Scotland  female       1\n",
       "4      HG00101        GBR  British in England and Scotland    male       1\n",
       "5      HG00102        GBR  British in England and Scotland  female       1\n",
       "6      HG00103        GBR  British in England and Scotland    male       1\n",
       "7      HG00105        GBR  British in England and Scotland    male       1\n",
       "8      HG00106        GBR  British in England and Scotland  female       1\n",
       "9      HG00107        GBR  British in England and Scotland    male       1\n",
       "10     HG00108        GBR  British in England and Scotland    male       1\n",
       "11     HG00109        GBR  British in England and Scotland    male       1\n",
       "12     HG00110        GBR  British in England and Scotland  female       1\n",
       "13     HG00111        GBR  British in England and Scotland  female       1\n",
       "14     HG00112        GBR  British in England and Scotland    male       1\n",
       "15     HG00113        GBR  British in England and Scotland    male       1\n",
       "16     HG00114        GBR  British in England and Scotland    male       1\n",
       "17     HG00115        GBR  British in England and Scotland    male       1\n",
       "18     HG00116        GBR  British in England and Scotland    male       1\n",
       "19     HG00117        GBR  British in England and Scotland    male       1\n",
       "20     HG00118        GBR  British in England and Scotland  female       1\n",
       "21     HG00119        GBR  British in England and Scotland    male       1\n",
       "22     HG00120        GBR  British in England and Scotland  female       1\n",
       "23     HG00121        GBR  British in England and Scotland  female       1\n",
       "24     HG00122        GBR  British in England and Scotland  female       1\n",
       "25     HG00123        GBR  British in England and Scotland  female       1\n",
       "26     HG00124        GBR  British in England and Scotland  female       1\n",
       "27     HG00125        GBR  British in England and Scotland  female       1\n",
       "28     HG00126        GBR  British in England and Scotland    male       1\n",
       "29     HG00127        GBR  British in England and Scotland  female       1\n",
       "...        ...        ...                              ...     ...     ...\n",
       "2504   NA21107        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2505   NA21108        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "2506   NA21109        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2507   NA21110        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "2508   NA21111        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2509   NA21112        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2510   NA21113        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2511   NA21114        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2512   NA21115        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2513   NA21116        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2514   NA21117        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2515   NA21118        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2516   NA21119        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2517   NA21120        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "2518   NA21122        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "2519   NA21123        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2520   NA21124        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2521   NA21125        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "2522   NA21126        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2523   NA21127        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2524   NA21128        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2525   NA21129        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2526   NA21130        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2527   NA21133        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2528   NA21135        GIH    Gujarati Indian in Houston,TX    male      26\n",
       "2529   NA21137        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "2530   NA21141        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "2531   NA21142        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "2532   NA21143        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "2533   NA21144        GIH    Gujarati Indian in Houston,TX  female      26\n",
       "\n",
       "[2534 rows x 5 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SampleID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'Sample'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-53-77d01ffd2635>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0mMut_Matrix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m'Mut'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m'Sample'\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minplace\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0mMut_Matrix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m \u001b[0mMut_Matrix2\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mMut_Matrix\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mSampleInfo\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Sample'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'inner'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m \u001b[0mMut_Matrix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/tools/merge.pyc\u001b[0m in \u001b[0;36mmerge\u001b[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy)\u001b[0m\n\u001b[1;32m     35\u001b[0m                          \u001b[0mright_on\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mright_on\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mleft_index\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mleft_index\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     36\u001b[0m                          \u001b[0mright_index\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mright_index\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msort\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msort\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msuffixes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msuffixes\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 37\u001b[0;31m                          copy=copy)\n\u001b[0m\u001b[1;32m     38\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     39\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0m__debug__\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/tools/merge.pyc\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, left, right, how, on, left_on, right_on, axis, left_index, right_index, sort, suffixes, copy)\u001b[0m\n\u001b[1;32m    181\u001b[0m         (self.left_join_keys,\n\u001b[1;32m    182\u001b[0m          \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mright_join_keys\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 183\u001b[0;31m          self.join_names) = self._get_merge_keys()\n\u001b[0m\u001b[1;32m    184\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/tools/merge.pyc\u001b[0m in \u001b[0;36m_get_merge_keys\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    350\u001b[0m                     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    351\u001b[0m                         \u001b[0mright_keys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 352\u001b[0;31m                     \u001b[0mleft_keys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mleft\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mlk\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    353\u001b[0m                     \u001b[0mjoin_names\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    354\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0m_any\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mleft_on\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/frame.pyc\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   1795\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1796\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1797\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1798\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1799\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/frame.pyc\u001b[0m in \u001b[0;36m_getitem_column\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   1802\u001b[0m         \u001b[0;31m# get column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1803\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_unique\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1804\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_item_cache\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1805\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1806\u001b[0m         \u001b[0;31m# duplicate columns & possible reduce dimensionaility\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/generic.pyc\u001b[0m in \u001b[0;36m_get_item_cache\u001b[0;34m(self, item)\u001b[0m\n\u001b[1;32m   1082\u001b[0m         \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcache\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1083\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mres\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1084\u001b[0;31m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1085\u001b[0m             \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_box_item_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1086\u001b[0m             \u001b[0mcache\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/internals.pyc\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, item, fastpath)\u001b[0m\n\u001b[1;32m   2849\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2850\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misnull\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2851\u001b[0;31m                 \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2852\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2853\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0misnull\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/index.pyc\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method)\u001b[0m\n\u001b[1;32m   1570\u001b[0m         \"\"\"\n\u001b[1;32m   1571\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mmethod\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1572\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_values_from_object\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1573\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1574\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/index.pyx\u001b[0m in \u001b[0;36mpandas.index.IndexEngine.get_loc (pandas/index.c:3824)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/index.pyx\u001b[0m in \u001b[0;36mpandas.index.IndexEngine.get_loc (pandas/index.c:3761)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'Sample'"
     ]
    }
   ],
   "source": [
    "## transpose matrix to put genotype on each row and add gender and populaton data for each sample\n",
    "\n",
    "Mut_MatrixRECODE.head()\n",
    "\n",
    "# attatch the population data to reference array MutMatrix\n",
    "Mut_Matrix.rename(columns={'':'Sample'}, inplace=True)\n",
    "Mut_Matrix.head()\n",
    "Mut_Matrix2=pd.merge(Mut_Matrix, SampleInfo, on=['Sample'], how='inner')\n",
    "Mut_Matrix.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'Mut'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-55-0ef81f45a479>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mprint\u001b[0m \u001b[0mMut_Matrix\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Mut'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/frame.pyc\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   1795\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1796\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1797\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1798\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1799\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_getitem_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/frame.pyc\u001b[0m in \u001b[0;36m_getitem_column\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   1802\u001b[0m         \u001b[0;31m# get column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1803\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_unique\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1804\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_item_cache\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1805\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1806\u001b[0m         \u001b[0;31m# duplicate columns & possible reduce dimensionaility\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/generic.pyc\u001b[0m in \u001b[0;36m_get_item_cache\u001b[0;34m(self, item)\u001b[0m\n\u001b[1;32m   1082\u001b[0m         \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcache\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1083\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mres\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1084\u001b[0;31m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1085\u001b[0m             \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_box_item_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1086\u001b[0m             \u001b[0mcache\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/internals.pyc\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, item, fastpath)\u001b[0m\n\u001b[1;32m   2849\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2850\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misnull\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2851\u001b[0;31m                 \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2852\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2853\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0misnull\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pandas/core/index.pyc\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method)\u001b[0m\n\u001b[1;32m   1570\u001b[0m         \"\"\"\n\u001b[1;32m   1571\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mmethod\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1572\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_values_from_object\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1573\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1574\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/index.pyx\u001b[0m in \u001b[0;36mpandas.index.IndexEngine.get_loc (pandas/index.c:3824)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/index.pyx\u001b[0m in \u001b[0;36mpandas.index.IndexEngine.get_loc (pandas/index.c:3761)\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'Mut'"
     ]
    }
   ],
   "source": [
    "print Mut_Matrix['Mut']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "Mut_Matrix.rename(columns = {'':'Sample'}, inplace = True)\n",
    "Mut_Matrix.head()\n",
    "\n",
    "## add enthnicity column\n",
    "## add disease column\n",
    "\n",
    "#add\n",
    "#FiltListMelt = pd.merge(FiltListMelt, SampleInfo['Populaton'], on=['Sample'], how='inner')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'Mut_Matrix' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-196-be6503ff8276>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mMut_Matrix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m''\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m'Sample'\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minplace\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mMut_Matrix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtail\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'Mut_Matrix' is not defined"
     ]
    }
   ],
   "source": [
    "Mut_Matrix.rename(columns = {'':'Sample'}, inplace = True)\n",
    "Mut_Matrix.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Transpose to make individual genomes listed on rows rather than columns\n",
    "Mut_Matrix=Mut_Matrix.transpose()\n",
    "        #add ethnicity\n",
    "#FiltListMelt = pd.merge(Mut_Matrix, SampleInfo['Population'], on=['Sample'], how='inner')\n",
    "#Output the data\n",
    "out='MatrixGenomDt'\n",
    "datasrc='Mt'\n",
    "fileNm = out + '_' + datasrc + '.csv'\n",
    "Mut_Matrix.to_csv(fileNm, sep=\"\\t\", index_label=\"False\", index=\"False\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "###format data - http://blog.nextgenetics.net/?e=44###"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM_x</th>\n",
       "      <th>POS_x</th>\n",
       "      <th>REF_x</th>\n",
       "      <th>ALT_x</th>\n",
       "      <th>INFO_x</th>\n",
       "      <th>Sample</th>\n",
       "      <th>value</th>\n",
       "      <th>Mut</th>\n",
       "      <th>CHROM_y</th>\n",
       "      <th>POS_y</th>\n",
       "      <th>...</th>\n",
       "      <th>ALT_y</th>\n",
       "      <th>INFO_y</th>\n",
       "      <th>Family ID</th>\n",
       "      <th>Population</th>\n",
       "      <th>Population Description</th>\n",
       "      <th>Gender</th>\n",
       "      <th>SnpDB</th>\n",
       "      <th>RSnum</th>\n",
       "      <th>Clinical_Significance</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MT</td>\n",
       "      <td>1438</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>VT=S;AC=2419</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>1</td>\n",
       "      <td>chrMT-1438-G</td>\n",
       "      <td>MT</td>\n",
       "      <td>1438</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>RS=2001030;RSPOS=1438;dbSNPBuildID=92;SSR=0;SA...</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/snp/2001030</td>\n",
       "      <td>2001030</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MT</td>\n",
       "      <td>15326</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>VT=S;AC=2521</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>1</td>\n",
       "      <td>chrMT-15326-G</td>\n",
       "      <td>MT</td>\n",
       "      <td>15326</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>RS=2853508;RSPOS=15326;dbSNPBuildID=100;SSR=0;...</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>male</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/snp/2853508</td>\n",
       "      <td>2853508</td>\n",
       "      <td>4</td>\n",
       "      <td>CYTB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MT</td>\n",
       "      <td>1438</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>VT=S;AC=2419</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>1</td>\n",
       "      <td>chrMT-1438-G</td>\n",
       "      <td>MT</td>\n",
       "      <td>1438</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>RS=2001030;RSPOS=1438;dbSNPBuildID=92;SSR=0;SA...</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/snp/2001030</td>\n",
       "      <td>2001030</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MT</td>\n",
       "      <td>15326</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>VT=S;AC=2521</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>1</td>\n",
       "      <td>chrMT-15326-G</td>\n",
       "      <td>MT</td>\n",
       "      <td>15326</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>RS=2853508;RSPOS=15326;dbSNPBuildID=100;SSR=0;...</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/snp/2853508</td>\n",
       "      <td>2853508</td>\n",
       "      <td>4</td>\n",
       "      <td>CYTB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MT</td>\n",
       "      <td>4216</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>VT=S;AC=139</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>1</td>\n",
       "      <td>chrMT-4216-C</td>\n",
       "      <td>MT</td>\n",
       "      <td>4216</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>RS=1599988;RSPOS=4216;dbSNPBuildID=88;SSR=1;SA...</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>GBR</td>\n",
       "      <td>British in England and Scotland</td>\n",
       "      <td>female</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/snp/1599988</td>\n",
       "      <td>1599988</td>\n",
       "      <td>0</td>\n",
       "      <td>ND1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM_x  POS_x REF_x ALT_x        INFO_x   Sample  value            Mut  \\\n",
       "0      MT   1438     A     G  VT=S;AC=2419  HG00096      1   chrMT-1438-G   \n",
       "1      MT  15326     A     G  VT=S;AC=2521  HG00096      1  chrMT-15326-G   \n",
       "2      MT   1438     A     G  VT=S;AC=2419  HG00097      1   chrMT-1438-G   \n",
       "3      MT  15326     A     G  VT=S;AC=2521  HG00097      1  chrMT-15326-G   \n",
       "4      MT   4216     T     C   VT=S;AC=139  HG00097      1   chrMT-4216-C   \n",
       "\n",
       "  CHROM_y  POS_y  ...  ALT_y  \\\n",
       "0      MT   1438  ...      G   \n",
       "1      MT  15326  ...      G   \n",
       "2      MT   1438  ...      G   \n",
       "3      MT  15326  ...      G   \n",
       "4      MT   4216  ...      C   \n",
       "\n",
       "                                              INFO_y Family ID Population  \\\n",
       "0  RS=2001030;RSPOS=1438;dbSNPBuildID=92;SSR=0;SA...   HG00096        GBR   \n",
       "1  RS=2853508;RSPOS=15326;dbSNPBuildID=100;SSR=0;...   HG00096        GBR   \n",
       "2  RS=2001030;RSPOS=1438;dbSNPBuildID=92;SSR=0;SA...   HG00097        GBR   \n",
       "3  RS=2853508;RSPOS=15326;dbSNPBuildID=100;SSR=0;...   HG00097        GBR   \n",
       "4  RS=1599988;RSPOS=4216;dbSNPBuildID=88;SSR=1;SA...   HG00097        GBR   \n",
       "\n",
       "            Population Description  Gender  \\\n",
       "0  British in England and Scotland    male   \n",
       "1  British in England and Scotland    male   \n",
       "2  British in England and Scotland  female   \n",
       "3  British in England and Scotland  female   \n",
       "4  British in England and Scotland  female   \n",
       "\n",
       "                                     SnpDB    RSnum Clinical_Significance  \\\n",
       "0  http://www.ncbi.nlm.nih.gov/snp/2001030  2001030                     2   \n",
       "1  http://www.ncbi.nlm.nih.gov/snp/2853508  2853508                     4   \n",
       "2  http://www.ncbi.nlm.nih.gov/snp/2001030  2001030                     2   \n",
       "3  http://www.ncbi.nlm.nih.gov/snp/2853508  2853508                     4   \n",
       "4  http://www.ncbi.nlm.nih.gov/snp/1599988  1599988                     0   \n",
       "\n",
       "   gene  \n",
       "0   NaN  \n",
       "1  CYTB  \n",
       "2   NaN  \n",
       "3  CYTB  \n",
       "4   ND1  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FiltListMelt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
